juju/juju
# Change grep pregex to regex

User noted that grep -P flag breaks compiling on macOS. Tested without the -P flag and the command `grep "^go"` works without the -P flag. Do not have MacOS so - would like someone with MacOS to test to see if that solves that issue. 

*Why this change is needed and what it does.*

## Checklist

*If an item is not applicable, use `~strikethrough~`.*


## QA steps

*Commands to run to verify that the change works.*

Change to a version of go that's not supported by go.mod and run "make install-dependencies"

If you get the message 

> required go version '1.19'

with the correct version in between the single quotes then it is parsing go.mod properly. 

```sh
$ sudo snap refresh go --channel=1.18/stable --classic
go (1.18/stable) 1.18.7 from Michael Hudson-Doyle (mwhudson) refreshed
$ make install-dependencies
Makefile:351: "warning: version of go too low: use 'snap refresh go --channel=1.19'"
Makefile:352: *** "error Installed go version '1.18.7' less than required go version '1.19'".  Stop.``````
